cloudigrade/account/util.py

```diff
@@ -126,80 +135,134 @@ def generate_aws_ami_messages(instances_data, ami_list):
     return messages
 
 
-def _create_exchange_and_queue(queue_name):
+def _get_sqs_queue_url(queue_name):
     """
-    Create a Kombu message Exchange and Queue.
+    Get the SQS queue URL for the given queue name.
+
+    This has the side-effect on ensuring that the queue exists.
 
     Args:
-        queue_name (str): The target queue's name
+        queue_name (str): the name of the target SQS queue
 
     Returns:
-        tuple(kombu.Exchange, kombu.Queue)
+        str: the queue's URL.
 
     """
-    exchange = kombu.Exchange(
-        settings.QUEUE_EXCHANGE_NAME,
-        'direct',
-        durable=True
-    )
-    message_queue = kombu.Queue(
-        queue_name,
-        exchange=exchange,
-        routing_key=queue_name
+    sqs = boto3.client('sqs')
+    try:
+        return sqs.get_queue_url(QueueName=queue_name)['QueueUrl']
+    except ClientError as e:
+        if e.response['Error']['Code'].endswith('.NonExistentQueue'):
+            return sqs.create_queue(QueueName=queue_name)['QueueUrl']
```

Comment:
Double failure?  Maybe log.

Reply:
This isn't a double failure. As weird as it may look, I think this is the best way for us to _ensure that a queue exists_ before cloudigrade can either read from or write to it. Frameworks like Kombu would do this for us (for reasonable queue backends, at least), but since we're using bare `boto3` calls and `boto3` itself doesn't auto-create queues when needed, we have to implement this pattern of manually checking on one, creating it if not present, and then returning its URL.

Reply:
I meant that creating a queue could result in a failure inside the except.

Reply:
How are catastrophic failures logged?  Do we provide context around what we were trying to do?